MSTR-169: Add JSONEditor library (#669)
2600hz/monster-ui: docs/monster-ui/jsoneditor().md

title: jsoneditor()

# monster.ui.jsoneditor()
The `monster.ui.jsoneditor()` method generates an instance of a JSON editor with a set of handy methods and properties.

![](images/jsoneditor-preview.png)

## Syntax
```javascript
monster.ui.jsoneditor(target[,options])
```

### Parameters

Key | Description | Type | Default | Required
:-: | --- | :-: | :-: | :-:

> `target` | A Jquery object which is automatically contain the JSON Editor | `jQuery` | |`true`

Reviewer: The "Type" column already provides the type of this parameter, no need to repeat yourself.
```suggestion
`target` | Container for the JSON Editor widget. | `jQuery` | |`true`
```

> `options` | Let you override default editor options that can be found on the [JSONEditor](https://github.com/josdejong/jsoneditor/blob/master/docs/api.md) page | ([#options](#options)) | |`false`

Reviewer: ```suggestion
`options` | Let you override default editor options that can be found on the [JSONEditor](https://github.com/josdejong/jsoneditor/blob/master/docs/api.md) page | `Object`([#options](#options)) | |`false`
```

### `options`

> Here is a list of the customization settings specific to Monster UI.

Reviewer: ```suggestion
List of helper specific customizable settings (in opposition to the ones exposed by the JSON Editor library).
```

Key | Description | Type | Default | Required
:-: | --- | :-: | :-: | :-:
`json` | Default JSON to set in the editor | `Object` | | `false`

### Return
An instance of the JSONEditor

## Description

> The `monster.ui.jsoneditor()` method renders a powerfull jsoneditor into the specified `target` which is completely customizable passing the `options` parameter. This method returns the `JSONEditor` instance so you have access to its properties and methods. Please see [JSONEditor](https://github.com/josdejong/jsoneditor) for reference. To get the JSONEditor instance from the DOM please see [getJsoneditor()](./getJsoneditor().md)

Reviewer: ```suggestion
The `monster.ui.jsoneditor()` method renders a customizable JSON editor into the specified `target`. This method returns the `JSONEditor` instance so you have access its properties and methods. Please see [JSONEditor](https://github.com/josdejong/jsoneditor) for reference. To get the JSONEditor instance from the DOM please see [getJsoneditor()](./getJsoneditor().md)
```

To initialize the JSONEditor, you just need to pass the `target` parameter:
```html

> <!--You must have to set the size of the container-->

Reviewer: ```suggestion
<!--You need to set the container's size -->
```

<div id="jsoneditor" style="width: 400px; height: 400px;"></div>
```

```javascript
var $target = $('#jsoneditor');

> var jsoneditor = monster.ui.jsoneditor($target)

Reviewer: ```suggestion
var jsoneditor = monster.ui.jsoneditor($target);
```
